### PR TITLE
Refresh ETCD worker list

### PR DESCRIPTION
**Pull request by ssyssy**
https://tachyonnexus.atlassian.net/browse/AC-1644

### CODE REVIEW

File: alluxio/alluxio_file_system.py [hunk 1]
```diff
@@ -137,6 +139,21 @@ def __init__(
             worker_addresses, self.logger
         )
         self.http_port = http_port
+        self.req_counter = 0
+        self.threadLock = threading.Lock()
+        def refresh_worker():
+            with self.threadLock:
+                if self.req_counter == 0:
```
Comment (LuQQiu) [inline alluxio/alluxio_file_system.py:146]: if there are always requests coming in, then we will never refresh worker list?
and also using a request counter may not always be correct. Some users may call partial of a method and then cancel

**Comment by LuQQiu** [comment]
TODO
- how to properly garbage collected the created daemon thread

**Comment by LuQQiu** [comment]
> TODO
> 
> * how to properly garbage collected the created daemon thread

Python does not have proper thread shutdown when instance cleanup.

Known limitation:
1. ETCD connections are controlled by Python process numbers. Each process will have a new Alluxio instance and one background thread pinging the membership service.
2. The ETCD background thread is long-running even if the Alluxio instance is cleaned or new Alluxio instance is created. Many redundant ETCD background threads may co-exist and left in the Python process.

**Comment by LuQQiu** [comment]
@ChunxuTang 我先merge, 你review完我继续改

**Comment by ChunxuTang** [comment]
Per discussion, it's probably better to use `async` instead of manual `threading`-based implementation due to the python GIL.

**Pull request merged**